docs/integration_pipeline_controller.md

```diff
@@ -0,0 +1,40 @@
+<div align="center"><h1>IntegrationPipeline Controller</h1></div>
+
+```mermaid
+
+%%{init: {'theme':'forest'}}%%
+flowchart TD
+  %% Defining the styles
+  classDef Red fill:#FF9999;
+  classDef Amber fill:#FFDEAD;
+  classDef Green fill:#BDFFA4;
+
+   %% Node definitions
+  predicate((PREDICATE: <br>integration pipeline <br> reconciliation))
+  get_resources{Get pipeline, <br> component, <br> & application}
+  report_status(Report status if Snapshot was created <br> for Pull requests)
+  check_tests{Check Snapshot <br> passed all tests}
+  check_supersede(Check if Snapshot requires <br> a composite supersede)  
```

Comment:
I'm thinking that this would be clearer as a decision point, maybe something like this:
```
Does Snapshot need to be <br> superseded with a composite Snapshot?
```
Which then leads to either updating the status or creating the composite Snapshot.

Reply:
done